core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java

```diff
@@ -311,6 +312,19 @@ public void format() throws IOException {
         .toString());
   }
 
+  /**
+   * Triggers a checkpoint in the primary master ufs journal.
+   *
+   * @return the snapshot response
+   */
+  public synchronized SnapshotPResponse checkpoint() {
+    if (mTailerThread == null) {
```

Comment:
isn't this normally `null` on the primary?

Reply:
Oh yeah, now change to directly `mMaster.writeToCheckpoint(journalWriter);` now.